core/extensions/impl/crypto_extension.cpp

```diff
@@ -168,4 +183,297 @@ namespace kagome::extensions {
 
     memory_->storeBuffer(out_ptr, common::Buffer(hash));
   }
+
+  // ---------------------- runtime api version 1 methods ----------------------
+
+  runtime::PointerSize CryptoExtension::ext_ed25519_public_keys_v1(
+      runtime::SizeType key_type) {
+    static const common::Buffer error_result{};
+
+    auto key_type_id = static_cast<crypto::KeyTypeId>(key_type);
+    if (!crypto::isSupportedKeyType(key_type_id)) {
+      auto kt = crypto::decodeKeyTypeId(key_type_id);
+      logger_->error("key type '{}' is not supported", kt);
+      return memory_->storeBuffer(error_result);
```

Comment:
probably need to crash instead of returning error, since spec say nothing about it

Reply:
Never chash manually. Instead this to use write down warn/err to Log and call AppStateManager::shutdown();